### PR TITLE
Catalog: bump effect 3.14.13 → 3.21.4

### PR DESCRIPTION
**Pull request by nl0**
Bumps `effect` from 3.14.13 to **3.21.4** (latest), covering the security advisory behind Renovate #4928.

**This is not a lockfile-only bump.** effect ≥3.18 changed JSON-Schema emission, Schema type inference, and fiber scheduling, so the bot PRs (#4779 @3.21.0, #4928 @3.20.0) — which touch only `package.json`/`package-lock.json` — would fail CI. This PR includes the required source + test fixes.

### Source fixes

| File | Change in effect ≥3.18 | Fix |
|------|------------------------|-----|
| `app/components/Assistant/Model/Tool.ts` | Empty-schema `$id` is now percent-encoded `/schemas/%7B%7D` (was the raw, draft-2020-invalid `/schemas/{}`), slipping past the sanitizer | Extend the condition to normalize both forms to `/schemas/empty` — restores the existing `navigation.spec` snapshot, no `-u` needed |
| `app/containers/Bucket/Routes.ts` | Curried `S.brand('S3Path')(S.String)` now infers a non-`never` schema `Context`, breaking `fromPathParams` | Switch to the idiomatic piped form `S.String.pipe(S.brand('S3Path'))` |
| `app/utils/Navigation.ts` | With the brand correctly propagating, a route's decoded params (branded) no longer match their string-record encoding | Widen `fromPathParams` to `<A, I>` and mark the identity `transformOrFail` non-strict (runtime-identical — brands are type-only) |

### Test fixes

- **`app/containers/Bucket/Routes.spec.ts` (new)** — round-trip guard for the real `s3Object`/`s3Prefix` routes over realistic S3 keys (spaces, `#`, `?`, non-ASCII, nested/trailing slashes), pinning the effect-Schema ↔ path-to-regexp encoding seam. Encoding is **per-segment** and `/` separators are preserved, so multi-segment keys still match `:path`.
- **`app/components/Assistant/Model/Conversation.spec.ts`** — effect 3.21's fiber scheduling change made two pre-existing specs hang, which ran the CI `test-catalog` job ~33 min before an out-of-memory crash. Both fixes are **test-only** (the Conversation actor and all product code are unchanged):
  - The LLM stub now signals it was invoked, then suspends instead of resolving synchronously — a real `converse` is an async round-trip, so the conversation stays in the observable `WaitingForAssistant` state. A synchronous resolve collapsed `Idle → WaitingForAssistant → Idle` so fast that `SubscriptionRef.changes` coalesced the transient state before any observer saw it.
  - A stub that re-issued the *same* `ToolUse` every round (an infinite `converse → ToolUse → ToolResult → converse` loop, previously cut short only by teardown) now returns the tool-use once, then settles. Under 3.21 the tight synchronous loop starved the observer and grew `events` without bound → the OOM.

### Verification

- `tsc --noEmit --skipLibCheck` clean (matches CI's `ForkTsCheckerWebpackPlugin` typecheck during `npm run build`).
- CI `test-catalog` green (~5 min; previously hung ~33 min → OOM before the Conversation fix). Full local suite passes apart from pre-existing Node-26-only `@testing-library/react-hooks` flakes in Athena Queries (unrelated; green on CI's Node 20).
- Routing round-trip baselined **byte-identical** against master + effect 3.14.13, confirming `strict: false` changes no runtime behavior and all three source fixes address genuine 3.21 regressions.

**Known boundary (pre-existing, unchanged by this PR):** a literal `%` in an S3 key isn't round-trip-safe — the URL→params path decodes twice (`path-to-regexp`'s matcher, then `S3PathFromString.decode`) against a single encode. Pinned as `it.fails` xfail tests so it surfaces the day the seam is fixed; tracked separately. This is the edge case Greptile originally flagged — see the resolved inline thread for the full analysis.

Supersedes #4779 and #4928.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<h3>Greptile Summary</h3>

This PR bumps `effect` and updates the affected catalog code paths.

- Upgrades `effect` from 3.14.13 to 3.21.4.
- Normalizes both old and new empty-schema `$id` formats.
- Updates branded S3 route schemas for newer Effect inference.
- Adds S3 object and prefix route round-trip tests.
- Adjusts assistant conversation tests for the newer scheduler behavior.

<h3>Confidence Score: 5/5</h3>

This looks safe to merge.

- No blocking issues found in the changed code.

<h3>Important Files Changed</h3>

| Filename | Overview |
|----------|----------|
| catalog/app/utils/Navigation.ts | Widens route param schema typing and keeps runtime path param encoding behavior aligned with the existing route flow. |
| catalog/app/containers/Bucket/Routes.ts | Switches the S3 path brand to the piped Effect Schema form used by the upgraded dependency. |
| catalog/app/containers/Bucket/Routes.spec.ts | Adds focused route round-trip coverage for S3 object and prefix paths. |
| catalog/app/components/Assistant/Model/Tool.ts | Extends empty-schema `$id` cleanup to handle the percent-encoded format emitted by newer Effect versions. |
| catalog/app/components/Assistant/Model/Conversation.spec.ts | Updates the test LLM stubs so observable conversation states are preserved under the newer scheduler. |
| catalog/package.json | Updates the direct `effect` dependency version. |
| catalog/package-lock.json | Locks the upgraded `effect` package version and integrity metadata. |

<sub>Reviews (2): Last reviewed commit: ["test(catalog): fix Conversation actor sp..."](https://github.com/quiltdata/quilt/commit/d10d66da8ac77aed5524c77b26a5e8957d4fba3f) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=40064961)</sub>

### CODE REVIEW

File: catalog/app/utils/Navigation.ts [hunk 1]
```diff
@@ -45,8 +45,16 @@ export const PathParams = S.Record({
...
+  A extends typeof PathParams.Type,
+  I extends typeof PathParams.Type,
+>(
+  schema: S.Schema<A, I>,
+) =>
   S.transformOrFail(PathParams, schema, {
+    // non-strict: decoded params may be branded/refined subtypes of the raw
+    // string-record encoding (e.g. S3Path), so the identity encode widens.
+    strict: false,
     encode: (toI) => Eff.Effect.succeed(toI),
```
[the remark this answers is not in the document]
Comment (nl0) [reply catalog/app/utils/Navigation.ts:58]: Verified empirically against the real `s3Object`/`s3Prefix` routes — the cited cases all round-trip correctly:

| key | encoded pathname |
|-----|------------------|
| `with space/and#hash/q?uery.txt` | `/b/b/tree/with%20space/and%23hash/q%3Fuery.txt` |
| `unicode/имя/файл.txt` | `…/unicode/%D0%B8%D0%BC%D1%8F/%D1%84%D0%B0%D0%B9%D0%BB.txt` |

`path-to-regexp`'s `compile` encodes **per segment** and preserves the `/` separators, so multi-segment keys still match `:path` — they don't collapse into one `%2F`-laden param. The identity `encode` here doesn't break that.

This is also not a behavior change from the effect bump: the same round-trip on `master` + effect 3.14.13 produces **byte-identical** output. `strict: false` is a type-level flag only.

There is one genuine pre-existing gap: a literal `%` in a key isn't round-trip-safe (the URL→params path decodes twice — `path-to-regexp`'s matcher, then `S3PathFromString.decode` — against a single encode). That's tracked separately and pinned as `it.fails` xfail tests in the new `Routes.spec.ts`, so it flips red the day the seam is fixed.

**Comment by nl0** [comment]
@greptileai review
